### ISSUE TITLE
Issues with List of Parcelables

### ISSUE THREAD

**Issue by lance0428** [mcharmas/android-parcelable-intellij-plugin#22]
I am having issues with List<Parceable> support.  I keep getting NullPointers when I try to read my object.  Below is the objects involed:

**Root object I am trying to parcel**

```java
public class MyObjectToStore implements Parcelable {
    public List<ObjectInList> objectsInList;
    public String myObjectValue;

    public MyObjectToStore() {
    }

    private MyObjectToStore(Parcel in) {
        in.readTypedList(objectsInList, ObjectInList.CREATOR);
        this.myObjectValue = in.readString();
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeTypedList(objectsInList);
        dest.writeString(this.myObjectValue);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<MyObjectToStore> CREATOR = new Creator<MyObjectToStore>() {
        public MyObjectToStore createFromParcel(Parcel source) {
            return new MyObjectToStore(source);
        }

        public MyObjectToStore[] newArray(int size) {
            return new MyObjectToStore[size];
        }
    };
}
```
-

**Inner Parcelable in list**
```java
public class ObjectInList implements Parcelable {
    String objectInListValue;

    private ObjectInList(Parcel in) {
        this.objectInListValue = in.readString();
    }

    public ObjectInList(String objectInListValue) {
        this.objectInListValue = objectInListValue;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(this.objectInListValue);
    }

    @Override
    public int describeContents() {
        return 0;
    }

    public static final Creator<ObjectInList> CREATOR = new Creator<ObjectInList>() {
        public ObjectInList createFromParcel(Parcel source) {
            return new ObjectInList(source);
        }

        public ObjectInList[] newArray(int size) {
            return new ObjectInList[size];
        }
    };
}
```

**Test that shows Exception**

```java
public class MyObjectToStoreTest extends TestCase {

    public void testParcelable() {
        MyObjectToStore objectToStore = new MyObjectToStore();
        objectToStore.myObjectValue = "MyObjectValue";
        List<ObjectInList> list = new ArrayList<ObjectInList>();
        list.add(new ObjectInList("Object1"));
        list.add(new ObjectInList("Object2"));
        objectToStore.objectsInList = list;

        Parcel parcel = Parcel.obtain();
        objectToStore.writeToParcel(parcel, 0);
        parcel.setDataPosition(0);

        MyObjectToStore fromParcel = MyObjectToStore.CREATOR.createFromParcel(parcel);
        assertEquals("MyObjectValue", fromParcel.myObjectValue);
        assertEquals(2, fromParcel.objectsInList.size());
        assertEquals("Object1", fromParcel.objectsInList.get(0).objectInListValue);
        assertEquals("Object2", fromParcel.objectsInList.get(1).objectInListValue);
    }
}
```

**Exception**

    java.lang.NullPointerException
    at android.os.Parcel.readTypedList(Parcel.java:1797)
    at objects.MyObjectToStore.<init>(MyObjectToStore.java:27)
    at objects.MyObjectToStore.<init>(MyObjectToStore.java:8)
    at objects.MyObjectToStore$1.createFromParcel(MyObjectToStore.java:33)
    at objects.MyObjectToStore$1.createFromParcel(MyObjectToStore.java:31)
    at objects.MyObjectToStoreTest.testParcelable(MyObjectToStoreTest.java:24)

It seems to blow up because the list in MyObjectToStore  is not being initialized properly by the generated code in *private MyObjectToStore(Parcel in)*. Shouldn't that constructor generate something like

```java
private MyObjectToStore(Parcel in) {
    objectsInList = new ArrayList<ObjectInList>(); // new
    in.readTypedList(objectsInList, ObjectInList.CREATOR);
    this.myObjectValue = in.readString();
}
```
Am I missing something? Is this an unimplemented feature, a bug, or a compromise because the code generator doesn't know what Type of list it is?

**Comment by zokipirlo** [comment]
I would suggest to use `createTypedArrayList`:
`objectsInList = in.createTypedArrayList(ObjectInList.CREATOR)`

**Comment by dallasgutauckis** [comment]
Agree 100%

**Comment by mcharmas** [comment]
This should be fixed in 0.6.2 (in review by JetBrains).
